### ISSUE TITLE
Top correlations index counts differ enormously from database content

### ISSUE THREAD

**Issue by chrisinmtown** [MISP/MISP#7512]
Please help me understand content of the Top correlations Index.  Get to this page by MISP -> Input Filters -> List Correlation Exclusions -> Top Correlations.  The counts shown there are totally different from what I see in the database.

To view this index, I click on the Regenerate cache button, wait, then when the data is fresh, it shows me 50 rows. I see the value `Postmaster notify: see transcript for details` in a fairly high spot, correlation count 5764.

I compared that result with result of this database query:

    select value, count(*) as count from correlations group by value order by count desc limit 20

The number one result there is value `Re:Pharmacy` with a count of 9,176,948!  That value does not appear on the Top correlations index page.  Digging further, I checked for the `Postmaster` value and found it occurs with a count of 7832.

So what is being counted here? Should I look for errors somewhere?  This page today is not useful.

### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | RedHat
| OS version (client)       | MacOS
| PHP version               | 7.4
| MISP version / git hash   | 2.4.143
| Browser                   | Firefox

### Expected behavior

The Top Correlations index should show me the most frequent data items. 

### Actual behavior

The Top Correlations index page shows me apparently random counts of frequently correlated values, definitely does not show the top ones.

**Comment by github-germ** [comment]
Check out function `generateTopCorrelations` in `Model/Correlation.php`

**Comment by chrisinmtown** [comment]
Looks like the Top correlations index queries Redis, not the database.  So if MISP has not cached data from a feed to Redis then this function cannot count the data.

**Comment by RichieB2B** [comment]
I can second this issue. The counts of `generateTopCorrelations` are way off from the counts in mysql. 
If you look at the function though it is setting up a new Redis table with name `CACHE_NAME` which is [defined ](https://github.com/MISP/MISP/blob/2.4/app/Model/Correlation.php#L10) as `misp:top_correlations`. This has nothing to do with the feeds which also can be cached in Redis.

**Comment by mokaddem** [comment]
Does the count in the index align with the output of the database if you regenerate the top correlation cache?
You can do it on the page `/correlations/top` by clicking the `Regenerate cache` button.

**Comment by RichieB2B** [comment]
No, it does not. Regenerating the correlations table was the first thing I did after noticing the difference.

**Comment by Nerlopo** [comment]
We've also seen this behaviour. Our top correlation from the database has a count of over 10m, whereas from the UI the same attribute has ~300k.

**Comment by JakubOnderka** [comment]
Thanks for report, it will be fixed in next MISP version.
